raiden/utils/filters.py

```diff
@@ -103,19 +112,47 @@ def __init__(self, web3: Web3, filter_params: dict):
         self._last_block: int = -1
         self._lock = Semaphore()
 
-    def get_new_entries(self, block_number: BlockSpecification):
+    def _do_get_new_entries(self, from_block: BlockSpecification, to_block: BlockSpecification):
+        filter_params = self.filter_params.copy()
+        filter_params['fromBlock'] = from_block
+        filter_params['toBlock'] = to_block
+
+        log.debug(
+            'Querying StatelessFilter',
+            from_block=from_block,
+            to_block=to_block,
+        )
+        result = self.web3.eth.getLogs(filter_params)
+        self._last_block = to_block
+        return result
+
+    def get_new_entries(self, target_block_number: BlockSpecification):
+        latest_block_number = self.web3.eth.blockNumber
         with self._lock:
-            filter_params = self.filter_params.copy()
-            filter_params['fromBlock'] = max(
-                filter_params.get('fromBlock', GENESIS_BLOCK_NUMBER),
+            result = []
+            from_block = max(
+                self.filter_params.get('fromBlock', GENESIS_BLOCK_NUMBER),
                 self._last_block + 1,
             )
-
-            if self.filter_params.get('toBlock') in (None, 'latest', 'pending'):
-                filter_params['toBlock'] = block_number or 'latest'
-
-            result = self.web3.eth.getLogs(filter_params)
-            self._last_block = filter_params.get('toBlock') or block_number
+            to_block = self.filter_params.get('toBlock')
+            if to_block in (None, 'latest', 'pending'):
+                if not isinstance(target_block_number, int):
```

Comment:
The behavior here looks different, previously it was forcing `toBlock` to be either the value given as a parameter or `latest`. Now, we preserve the `toBlock` as it is unless the `target_block_number` is an actual number, was this intentional?

Reply:
Humm, I think my explanation above is incorrect, let see if I get right this time.

Previously, if `self.filter_params` had an undefined range, or `latest`, or `pending`, then the user argument was considered. I believe this was done to forbid a second from going over the `toBlock` range set in the filter constructor.

Okay, so it seems that the behavior is the same.

Reply:
yes behaviour should be the same